src/main/java/facades/SearchFacade_Impl.java

```diff
@@ -245,13 +248,9 @@ public PersonDTO_OUT deletePerson(Integer id) {
             em.getTransaction().begin();
             // Find Person from ID
             Person person = em.find(Person.class, id);
-            if (person == null) {
+            if (person == null || person.getAddress() != null) {
                 throw new WebApplicationException("Could not delete Person. Provided ID does not exist.", 400);
             }
```

Comment:
I'm not sure I understand the change here. Can you explain it?

Reply:
Du får lige forklaringen alligevel, i tilfælde af at der er noget der kan misforstås:
I vores deleteCity metode der valgte vi at sige at vi kun kan delete en City hvis der ikke er nogle Address-referencer tilknyttet. 
For at kunne gøre det samme med Person og Address er det nødvendigt med et specifikt tjek og kast af exception (da constraints på database niveau egentlig fint tillader at vi sletter Person selvom der er er addresser på). . 

Af andre løsninger er der:
- Lade der være okay at slette en Person selvom der er en Address-reference
- Lave et tjek for om der er andre Person-objekter med samme Address-reference, og tillade sletning hvis der ikke findes andre. 

Sidste løsning synes jeg selv bedst om, men da der ikke var lagt op til noget i den stil i metode-implementationen i forvejen, så valgte jeg middelvejen og ville lige høre hvad i andre synes om det :)

Reply:
Cool, tak for den detaljerede gennemgang.

Jeg har også haft mig nogle overvejelser om database opsætningen ifbm. med phone.number. Denne burde være unik.

I forbindelse med dette samt ovenstående, så tror jeg virkelig vi kan gøre meget med JPA som vi ikke udnytter, såsom at gøre dette nemmere for os.